src/Unit_cell/atom_type.h

```diff
@@ -34,852 +34,868 @@
-                TERMINATE("wrong number of radial points");
-            }
-            free_atom_radial_grid_ = Radial_grid_ext<double>(num_points__, points__);
-        }
+    inline std::string const& label() const
+    {
+        return label_;
+    }
 
-        inline void set_free_atom_density(int num_points__, double const* dens__)
-        {
-            free_atom_density_.resize(num_points__);
-            std::memcpy(free_atom_density_.data(), dens__, num_points__ * sizeof(double));
-        }
+    inline std::string const& file_name() const
+    {
+        return file_name_;
+    }
 
-        inline Spline<double> const& beta_rf(int idxrf__) const
-        {
-            return beta_rf_[idxrf__];
-        }
+    inline int offset_lo() const
+    {
+        assert(offset_lo_ >= 0);
+        return offset_lo_;
+    }
+
+    inline void set_d_mtrx_ion(matrix<double>& d_mtrx_ion__)
+    {
+        pp_desc_.d_mtrx_ion = matrix<double>(d_mtrx_ion__.size(0), d_mtrx_ion__.size(1));
+        d_mtrx_ion__ >> pp_desc_.d_mtrx_ion;
+    }
+
+    inline mdarray<int, 2> const& idx_radial_integrals() const
+    {
+        return idx_radial_integrals_;
+    }
 
-        inline Spline<double> const& q_rf(int idx__, int l__) const
-        {
-            return q_rf_(idx__, l__);
+    inline mdarray<double, 3>& rf_coef() const
+    {
+        return rf_coef_;
+    }
+
+    inline mdarray<double, 3>& vrf_coef() const
+    {
+        return vrf_coef_;
+    }
+
+    inline Simulation_parameters const& parameters() const
+    {
+        return parameters_;
+    }
+
+    inline void set_free_atom_radial_grid(int num_points__, double const* points__)
+    {
+        if (num_points__ <= 0) {
+            TERMINATE("wrong number of radial points");
         }
+        free_atom_radial_grid_ = Radial_grid_ext<double>(num_points__, points__);
+    }
+
+    inline void set_free_atom_density(int num_points__, double const* dens__)
+    {
+        free_atom_density_.resize(num_points__);
+        std::memcpy(free_atom_density_.data(), dens__, num_points__ * sizeof(double));
+    }
+
+    inline double_complex f_coefficients(int xi1, int xi2, int s1, int s2) const
+    {
+        return f_coefficients_(xi1, xi2, s1, s2);
+    }
+    inline Spline<double> const& beta_rf(int idxrf__) const
+    {
+        return beta_rf_[idxrf__];
+    }
+
+    inline Spline<double> const& q_rf(int idx__, int l__) const
+    {
+        return q_rf_(idx__, l__);
+    }
+
+    /// compare the angular, total angular momentum and radial part of
+    /// the beta projectors, leaving the m index free. Only useful
+    /// when spin orbit coupling is included.
+    inline bool compare_index_beta_functions(const int xi, const int xj) const
+    {
+        return ((indexb(xi).l == indexb(xj).l) && (indexb(xi).idxrf == indexb(xj).idxrf) &&
+                (fabs(indexb(xi).j - indexb(xj).j) < 1e-8));
+    }
+
+  private:
+    void Generate_F_coefficients(void);
```

Comment:
methods in general start with small letter 
Generate_F_coefficients -> generate_F_coefficients
Calculate_U_sigma_m -> calculate_U_sigma_m